lib/SPIRV/libSPIRV/SPIRVModule.cpp

```diff
@@ -2399,7 +2411,10 @@ std::istream &operator>>(std::istream &I, SPIRVModule &M) {
     return MI.parseSPT(I);
   }
 #endif
-  return MI.parseSPIRV(I);
+  if (!MI.TranslationOpts.isReport()) {
+    return MI.parseSPIRV<false>(I);
```

Comment:
Can we not pass the isReport as an argument? Templated code might result in unneeded code bloating here.

Thanks

Reply:
Unfortunately, at this stage no. We don't want to introduce unneeded branch in parser hot path for both report and binary cases. User using binary translation shouldn't pay price for Reporting API. This could be avoided if the implementation wouldn't be in iostreams. That makes efficient implementation of early stop very hard.
In this case although you have right that the code size will increase due to two implementations of the same function but prom performance PoV shouldn't make any difference. During first enter icache would be cold anyways and then it's circulating in the same hot routines. There won't be a situation when you have both `<true>` and `<false>` cases loaded to icache in the same time so really problem is negligible. What's more this template can be removed during removal of iostreams as main parsing system.